charts/backstage/templates/_helpers.tpl

```diff
@@ -21,3 +21,25 @@ Returns a secret name for service to service auth
         {{- include "common.names.fullname" . -}}-auth
     {{- end -}}
 {{- end -}}
+
+{{/*
+Sets the secretKeyRef name for Backstage to the PostgreSQL existing secret if it present
+*/}}
+{{- define "janus-idp.backstage.existingSecret" -}}
+    {{- if .Values.postgresql.auth.existingSecret -}}
```

Comment:
Postgresql subchart can take these from global scope as well.

`global.postgresql.auth.existingSecret`

https://github.com/bitnami/charts/tree/main/bitnami/postgresql

Reply:
Awesome, thanks for the heads up!

Reply:
Updated with your suggestion, I had to add `global.postgresql.auth.existingSecret` to the values to ensure `helm template` doesn't throw an error. 

Also, I found during testing that the [backstage deployment](https://github.com/backstage/charts/blob/main/charts/backstage/templates/_helpers.tpl#L48) does not honor the `global.postgresql.auth.existingSecret` in the upstream chart. So, that will probably need to be updated after this.